### PR TITLE
Take the page type into account for caching - Part 1

### PR DESCRIPTION
**Pull request by guelzow**
Hi everyone.

Please review, test and comment. 😃 

We retrieve the page type via the global request object, if available.
This object is used all over the TYPO3 core, so I guess this is OK'ish.

Tobi

### CODE REVIEW

**Comment by aimeos** [comment]
After thinking about this for a while, there may be a better alternative esp. because the config array passed to the cache adapter is deprecated as you've already noticed.

The `getBody()` and `getHeader()` methods accept a parameter to distinguish different plugin instances and this could be used for the types as well:
https://github.com/aimeos/aimeos-typo3/blob/master/Classes/Controller/AbstractController.php#L164-L166
e.g.:
```
return $client->getBody( $pageType );
```

File: Classes/Controller/AbstractController.php [hunk 1]
```diff
@@ -163,7 +163,14 @@ protected function getClientOutput( \Aimeos\Client\Html\Iface $client ) : string
 
 		$this->response->addAdditionalHeaderData( (string) $client->getHeader() );
```
Comment (aimeos) [inline Classes/Controller/AbstractController.php:164]: You need the page type for the header too, otherwise you will get mixed header/body combinations

Comment (guelzow) [reply Classes/Controller/AbstractController.php:164]: True.
The header links (like the canonical for example) then would get mixed up.

**Comment by aimeos** [comment]
Thanks a lot

**Comment by guelzow** [comment]
Thanks for the patience (and the review).
Took me a while to start the pull request.

**Pull request merged**